block/block.md

```diff
@@ -1 +1,99 @@
+
+#### About Soft/Hard Confirmations
+
+The block manager retrieves blocks from both the p2p network and the underlying DA network because the blocks are available in the p2p network faster and DA retrieval is slower (e.g., 1 second vs 15 seconds). The blocks retrieved from the p2p network are only marked as soft confirmed until the DA retrieval succeeds on those blocks and they are marked hard confirmed. The hard confirmations can be considered to have a higher level of finality.
+
+### Block Retrieval from BlockStore (P2P BlockSync)
+
+The block manager of the full nodes regularly pulls blocks from the block store (which in turn uses the p2p network for syncing the blocks) at `BlockTime` intervals and starts off with a block store height of zero. Every time the block store height is higher than the last seen height, the newest blocks are pulled from the block store and sent to be applied (or state update), along with updating the last seen block store height.
+
+### Block Syncing
+
+The block manager stores and applies the block every time a new block is retrieved either via the blockstore or DA network. Block syncing involves:
+
+* `ApplyBlock` using executor: validate the block, executes the block (applies the transactions), captures the validator updates, and creates an update state.
+* `Commit` using executor: commit the execution and changes, update mempool, and publish events
+* Store the block, the validators, and the updated state.
+
+## Message Structure/Communication Format
+
+The communication between the block manager and executor:
+
+* `InitChain`: using the genesis, a set of parameters, and validator set to invoke `InitChainSync` on the proxyApp to obtain initial `appHash` and initialize the state.
+* `Commit`: commit the execution and changes, update mempool, and publish events.
+* `CreateBlock`: prepare a block by polling transactions from mempool.
+* `ApplyBlock`: validate the block, execute the block (apply transactions), validator updates, create and return updated state
+
+The communication between the full node and block manager:
+
+* Notify when the block is published
+* Notify when the block is done lazy building
+
+## Assumptions and Considerations
+
+* The block manager loads the initial state from the local store and uses genesis if not found in the local store, when the node (re)starts.
```

Comment:
I would add what it means to have a successful submission to the da-layer and what it does not mean.

Reply:
i think you are talking about the tx validity rules. since this is performed in the mempool pkg, we should defer it to mempool specs. once the tx is validated and lands in the mempool, it is picked up by the block manager to compose a block, etc. the possible failure scenarios afterwards would be ApplyBlock and SubmitBlockToDA. The specs for ApplyBlock should be covered in the executor specs and SubmitBlockToDA scenario should be covered in the DA specs.

Reply:
I was talking about a Celestia Transaction. A rollup Block i will edit my comment, cc @S1nus for mempool pkg @tuxcanfly for celestia, @Manav-Aggarwal for executor and @tzdybal for DA-Interface. Let's try figure this out.

Reply:
cc @gupadhyaya I do believe Manager should be aware of those interactions as we might be making the wrong assumptions here

Reply:
created https://github.com/rollkit/rollkit/issues/1236